### PR TITLE
Issue #383: Update implementation on add feature in DG

### LINKED ISSUE #383: Update DG - Implementation `add` feature

### PR DESCRIPTION
**Pull request by domlimm**
Closes #383.

### CODE REVIEW

File: docs/diagrams/AddActivityDiagram.puml [hunk 1]
```diff
@@ -4,9 +4,14 @@
 start
 :User executes add command;
 
-if () then ([candidate does not exist in system])
-    :Save candidate to system;
-else ([else])
+if () then ([valid command])
```
Comment (charmainehly) [inline docs/diagrams/AddActivityDiagram.puml:7]: 👍

**Review by charmainehly — CHANGES_REQUESTED** [review]
LGTM dom!! Added a comment on one of the areas you caught, not sure if you think it would be good to have a disclaimer/note at the top of the dg!! - if not just lmk and ill merge this! 👍

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -177,9 +177,9 @@ List of new entities:
 ### Add feature
 
 #### What is the feature about?
-The `add` mechanism is facilitated by `AddressBook`. The implementation of adding a `Candidate` through the `add` command has been enhanced on the existing approach. It extends `Command`. The input parameters of the `add` command has been tweaked slightly, introducing a few more related entities (Refer to [TODO: UPDATE LINK]).
+The `add` mechanism is facilitated by `AddressBook` (TAlent Assistant™'s development team has decided to stick with AB3 naming convention on this part). The implementation of adding a `Candidate` through the `add` command has been enhanced on the existing approach. It extends `Command`. The input parameters of the `add` command has been tweaked, introducing a few more related attributes (Refer to the [Current Implementation](#current-implementation)).
```
[the remark this answers is not in the document]
Comment (domlimm) [reply docs/DeveloperGuide.md:180]: Yes, let me do that! Good idea! 👍🏻 👍🏻 👍🏻

[the remark this answers is not in the document]
Comment (domlimm) [reply docs/DeveloperGuide.md]: @charmainehly just made the changes!

**Review by charmainehly — APPROVED** [review]
lgtm thank you!

**Pull request merged**